From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Baocheng Su <baocheng.su@siemens.com>
Date: Wed, 3 Jan 2024 21:27:22 +0800
Subject: [PATCH] mraa-gpio: pwm: Set pin options according to board type

This keeps the UI the same as before 25d7d66 for board types other than
IOT2050.

Fixes: 25d7d668ef11 ("Add IOT2050 board into mraa Intel nodes (#940)")
Signed-off-by: Baocheng Su <baocheng.su@siemens.com>
---
 hardware/intel/mraa-gpio-pwm.html | 28 ++++++++++++----------------
 1 file changed, 12 insertions(+), 16 deletions(-)

diff --git a/hardware/intel/mraa-gpio-pwm.html b/hardware/intel/mraa-gpio-pwm.html
index 2ec79961fc83..24bb095fefcd 100644
--- a/hardware/intel/mraa-gpio-pwm.html
+++ b/hardware/intel/mraa-gpio-pwm.html
@@ -25,6 +25,18 @@
                 return this.name?"node_label_italic":"";
             },
             oneditprepare: function() {
+                if (RED.settings.mraaGpioPwmBoardType === "IOT2050") {
+                    for (let i = 4; i <= 9; i++) {
+                        $('#node-input-pin').append($("<option></option>").attr("value", i).text("D" + i));
+                    }
+                } else {
+                    $('#node-input-pin').append($("<option></option>").attr("value", 3).text("D3"));
+                    $('#node-input-pin').append($("<option></option>").attr("value", 5).text("D5"));
+                    $('#node-input-pin').append($("<option></option>").attr("value", 6).text("D6"));
+                    $('#node-input-pin').append($("<option></option>").attr("value", 9).text("D9"));
+                    $('#node-input-pin').append($("<option></option>").attr("value", 10).text("D10"));
+                    $('#node-input-pin').append($("<option></option>").attr("value", 11).text("D11"));
+                }
                 $('#node-input-pin').val(this.pin);
                 $('#type-tip').text(RED.settings.mraaGpioPwmBoardType);
                 $('#ver-tip').text(RED.settings.mraaGpioPwmMraaVersion);
@@ -48,22 +60,6 @@
         <label for="node-input-pin"><i class="fa fa-circle"></i> Pin</label>
         <select type="text" id="node-input-pin" style="width: 250px;">
             <option value='' disabled selected style='display:none;'><span data-i18n="rpi-gpio.label.selectpin"></span></option>
-	    <optgroup label="Intel Galileo/Edison">
-                <option value="3">D3</option>
-                <option value="5">D5</option>
-                <option value="6">D6</option>
-                <option value="9">D9</option>
-                <option value="10">D10</option>
-                <option value="11">D11</option>
-            </optgroup>
-            <optgroup label="Siemens IOT2050">
-                <option value="4">D4</option>
-                <option value="5">D5</option>
-                <option value="6">D6</option>
-                <option value="7">D7</option>
-                <option value="8">D8</option>
-                <option value="9">D9</option>
-            </optgroup>
        </select>
     </div>
     <div class="form-row">
